biz.aQute.bnd.reporter/src/biz/aQute/bnd/reporter/plugins/entries/bundle/GogoPlugin.java

```diff
@@ -0,0 +1,375 @@
+
+						}
+						if (!propertiesName.contains(pName)) {
+
+							if (property.hasAttribute(VALUE_ATTR)) {
+								String val = property.getAttribute(VALUE_ATTR);
+
+								if (SCOPE.equals(pName)) {
+									scope = val;
+								}
+								if (FUNCTION.equals(pName)) {
+									functions = Arrays.asList(val);
+								}
+							} else {
+
+								final String[] vals = property.getTextContent()
+									.split("\n");
+
+								if (SCOPE.equals(pName)) {
+									if (vals.length == 1) {
+										scope = vals[0];
+									} else {
+
+										_reporter.warning(
+											"More than one gogo scope set as property name attribute for component %s",
+											path);
+									}
+
+								}
+								if (FUNCTION.equals(pName)) {
+									functions = Arrays.asList(vals);
+								}
+
+							}
+						}
+
+					}
+					if (scope == null || functions == null || functions.isEmpty()) {
+
+						continue;
+					}
+					final String fScope = scope;
+					Optional<GogoScopeDTO> optional = gogos.stream()
+						.filter(dto -> fScope.equals(dto.title))
+						.findFirst();
+
+					GogoScopeDTO gogoScopeDTO = null;
+					if (optional.isPresent()) {
+						gogoScopeDTO = optional.get();
+
+					} else {
+						gogoScopeDTO = new GogoScopeDTO();
+						gogoScopeDTO.title = scope;
+						gogoScopeDTO.functions = new ArrayList<GogoFunctionDTO>();
+
+					}
+
+					for (final String function : functions) {
+
+						Optional<GogoFunctionDTO> optionalF = gogoScopeDTO.functions.stream()
+							.filter(dto -> function.equals(dto.title))
+							.findFirst();
+
+						GogoFunctionDTO gogoFunctionDTO = null;
+
+						if (optionalF.isPresent()) {
+							gogoFunctionDTO = optionalF.get();
+						} else {
+							gogoFunctionDTO = new GogoFunctionDTO();
+							gogoFunctionDTO.title = function;
+							gogoFunctionDTO.methods = new ArrayList<GogoMethodDTO>();
+						}
+						String pathClazz = implementationClass.replace(".", "/") + ".class";
+						Resource resClazz = jar.getResource(pathClazz);
+						Analyzer analyzer = new Analyzer();
+						Clazz clazz = new Clazz(analyzer, null, resClazz);
+						clazz.parseClassFile(resClazz.openInputStream());
+						gogoFunctionDTO.methods.addAll(createMethodDTO(clazz, function));
+
+						gogoScopeDTO.functions.add(gogoFunctionDTO);
+					}
+					gogos.add(gogoScopeDTO);
+				}
+			} else {
+				if (!path.contains("*")) {
+					_reporter.warning("xml component file not found at path %s", path);
+				}
+			}
+		}
+		return !gogos.isEmpty() ? gogos : null;
+	}
+
+	private Collection<GogoMethodDTO> createMethodDTO(final Clazz clazz, final String function) throws Exception {
+
+		clazz.parseClassFile();
+
+		return clazz.methods()
+			.filter(m -> function.equals(m.getName()))
+			.map(m -> {
+
+				GogoMethodDTO gogoMethodDTO = new GogoMethodDTO();
+				gogoMethodDTO.title = m.getName();
+				gogoMethodDTO.parameters = Lists.newArrayList();
+
+				Optional<Annotation> oDescriptorAnnotation = m.annotations("*")
```

Comment:
I am unsure why you pass `*` as a the name of the annotation you want and then filter it later to the actual name you want. Seems like you should just pass the annotation name you want instead of `*` here.

Reply:
I first tryd "org.apache.felix.service.command.Descriptor" but that didn't work. I didn't know that binary name is "org/apache/felix/service/command/Descriptor".
done. thx